alpa/pipeline_parallel/computation.py

```diff
@@ -303,66 +303,48 @@ def donate_intermediates(self, computation):
         for invar in donated_invars:
             self.donated_invars[var_indices[invar]] = True
 
-    def get_compiled(self, mesh=None, is_distributed=None):
-        """Compile the XLA computation to get sharding specs."""
-        # TODO(yonghao): use more general cache functions
-        if self.compiled is not None:
-            return self.compiled
-
-        if (not isinstance(mesh, PhysicalDeviceMesh) and
-                is_distributed is None):
-            raise RuntimeError(
-                "Require a pre-allocated physical mesh to compile the runnable."
-            )
-
-        is_distributed = (mesh.is_distributed
-                          if mesh is not None else is_distributed)
+    def get_spmd_partitioned(self):
```

Comment:
We only need `num_devices` to run the spmd partitioner, because the sharding annotations are already in the HLO module. `num_devices` can be got from `self.strategy_config.logical_mesh_shape`